src/test/java/co/cask/hydrator/salesforce/parser/SalesforceQueryParserTest.java

```diff
@@ -13,76 +13,77 @@
-
-    queries.put("SELECT LeadSource, COUNT(Name) FROM Lead GROUP BY LeadSource",
-                new Object[]{"lead", new String[]{"LeadSource", "COUNT(Name)"}});
-
-    queries.put("SELECT Name, COUNT(Id) FROM Account GROUP BY Name HAVING COUNT(Id) > 1",
-                new Object[]{"account", new String[]{"Name", "COUNT(Id)"}});
+                new Object[]{"Account", new String[]{"Name"}});
 
     queries.put("SELECT Name, Id FROM Merchandise__c ORDER BY Name OFFSET 100",
-                new Object[]{"merchandise__c", new String[]{"Name", "Id"}});
+                new Object[]{"Merchandise__c", new String[]{"Name", "Id"}});
 
     queries.put("SELECT Name, Id FROM Merchandise__c ORDER BY Name LIMIT 20 OFFSET 100",
-                new Object[]{"merchandise__c", new String[]{"Name", "Id"}});
+                new Object[]{"Merchandise__c", new String[]{"Name", "Id"}});
 
     queries.put("SELECT Id, Name, Account.Name FROM Contact WHERE Account.Industry = 'media'",
-                new Object[]{"contact", new String[]{"Id", "Name", "Account.Name"}});
+                new Object[]{"Contact", new String[]{"Id", "Name", "Account.Name"}});
 
     queries.put("SELECT Id, FirstName__c, FirstName__c FROM Daughter__c WHERE Mother_of_Child__r.LastName__c LIKE 'C%'",
-                new Object[]{"daughter__c", new String[]{"Id", "FirstName__c", "FirstName__c"}});
+                new Object[]{"Daughter__c", new String[]{"Id", "FirstName__c", "FirstName__c"}});
 
     queries.put("SELECT Id, Who.FirstName, Who.LastName FROM Task WHERE Owner.FirstName LIKE 'B%'",
-                new Object[]{"task", new String[]{"Id", "Who.FirstName", "Who.LastName"}});
+                new Object[]{"Task", new String[]{"Id", "Who.FirstName", "Who.LastName"}});
 
     queries.put("SELECT UserId, LoginTime from LoginHistory",
-                new Object[]{"loginhistory", new String[]{"UserId", "LoginTime"}});
+                new Object[]{"LoginHistory", new String[]{"UserId", "LoginTime"}});
+
+    queries.put("SELECT LoginHistory.UserId, LoginHistory.LoginTime from LoginHistory",
+      new Object[]{"LoginHistory", new String[]{"UserId", "LoginTime"}});
+
+    queries.put("SELECT lh.UserId, lh.LoginTime from LoginHistory lh",
+      new Object[]{"LoginHistory", new String[]{"UserId", "LoginTime"}});
 
-    queries.put("SELECT UserId, COUNT(Id) from LoginHistory WHERE LoginTime > " +
-                  "2010-09-20T22:16:30.000Z AND LoginTime < 2010-09-21T22:16:30.000Z GROUP BY UserId",
-                new Object[]{"loginhistory", new String[]{"UserId", "COUNT(Id)"}});
-  };
+    queries.put("SELECT lh.UserId, lh.LoginTime from LoginHistory AS lh",
+      new Object[]{"LoginHistory", new String[]{"UserId", "LoginTime"}});
+
+    queries.put("SELECT UserId from LoginHistory WHERE LoginTime > " +
+                  "2010-09-20T22:16:30.000Z AND LoginTime < 2010-09-21T22:16:30.000Z",
+                new Object[]{"LoginHistory", new String[]{"UserId"}});
+  }
```

Comment:
add some tests for nested queries?

Reply:
Such cases are tested in the `testQueryParseError()`.